protocol-units/bridge/contracts/src/AtomicBridgeInitiatorMOVE.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {IAtomicBridgeInitiatorMOVE} from "./IAtomicBridgeInitiatorMOVE.sol";
+import {MockMOVEToken} from "./MockMOVEToken.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+contract AtomicBridgeInitiatorMOVE is IAtomicBridgeInitiatorMOVE, OwnableUpgradeable {
+    enum MessageState {
+        INITIALIZED,
+        COMPLETED,
+        REFUNDED
+    }   
+
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        bytes32 recipient;
+        bytes32 hashLock;
+        uint256 timeLock; // in blocks
+        MessageState state;
+    }
+
+    // Mapping of bridge transfer ids to BridgeTransfer structs
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+
+    // Total WETH pool balance
+    uint256 public poolBalance;
```

Comment:
Wouldn't an internal state variable be cheaper than external calls? 

@0xmovses @musitdev want to weigh in on this?

For accuracy it may be better to do `moveToken.balanceOf(address(this))`

but the math is simple enough that maybe the internal state variable `poolBalance` is nicer for readability?

Reply:
Let's capture Primata's suggestion in an issue and do the gas checks at a later time. For now let's keep parity with the WETH implementation.